File: solutions/python/bob/1/bob.py
def response(hey_bob):
    caracteres_a_eliminar = [" ", "\t", "\n", "\r"]
    for caracter in caracteres_a_eliminar:
        hey_bob = hey_bob.replace(caracter, "")
    if hey_bob =="":
        return "Fine. Be that way!"
    if tuple(hey_bob[-1]) == ("?",) and hey_bob.isupper():
        return "Calm down, I know what I'm doing!"
    elif hey_bob.isupper():
        return "Whoa, chill out!"
    elif tuple(hey_bob[-1]) == ("?",):
        return "Sure."
    
    else:
        return "Whatever."

File: solutions/scala/bob/1/src/main/scala/Bob.scala
object Bob {
  def response(statement: String): String =

    val cleanStatement: String = statement.trim

    def esSilencio(s: String): Boolean = s.length == 0
    def esGrito(s: String): Boolean =
      val letras = s.filter(_.isLetter)
      letras.nonEmpty && letras.forall(_.isUpper)
    def esPregunta(s: String): Boolean = s.last == '?'

    cleanStatement match {
      case s if esSilencio(s) => "Fine. Be that way!"
      case s if esGrito(s) && esPregunta(s) =>
        "Calm down, I know what I'm doing!"
      case s if esGrito(s)    => "Whoa, chill out!"
      case s if esPregunta(s) => "Sure."
      case _                  => "Whatever."
    }
}
